File: cifrado/escritorio/codigo/Kotlin-Fx-Tornado/prueba2/src/main/kotlin/com/example/demo/app/Cesar.kt
package com.example.demo.app

class Cesar
{
    var palabra = ""
    var salto = 0
    var resultado = ""
    
    fun preCifrado( salto:Int )
    {
        this.salto = salto
    }
    fun cifrar()
    {
        resultado = crear( palabra ,salto )
    }

    fun crear(orig:String,avanze:Int):String
    {
        var nuevo = ""
        for(letra in orig)
        {
            var aux = Character.toString(letra)
            nuevo += sustituir(aux, crearAlfabeto(), avanze)
        }
        return nuevo
    }

    fun sustituir(letra:String,lista:ArrayList<String>,avanze:Int):String
    {
        var indice = lista.indexOf(letra)
        var tam = lista.size
        if( indice + avanze >= tam)
            {return lista[(indice - tam) + avanze]}
        return lista[indice + avanze]
    }

    fun crearAlfabeto():ArrayList<String>
    {
        val list: ArrayList<String> = ArrayList()
        
        var caract: Char
        caract = ' '
        while (caract <= 'Ñ') {
            list.add(Character.toString(caract))
            ++caract
        }
        return list
    }

}

File: cifrado/web/codigo/Python-Nagare/Counter/Counter/cesar.py

class Cesar:

    def __init__(self ):
        self.palabra = ""
        self.salto = 0
        self.resultado = "..."
        self.alfabeto = self.Crearalfabeto()

    def cifrar(self):
        self.resultado = self.proceso( self.palabra ,self.salto )
        
    def proceso(self,orig, cont):
        e = self.alfabeto
        return "".join(list(map(lambda x: self.sustituir(x, e, cont), list(orig))))

    def sustituir(self,letra, lista, avanze):
        indice = lista.index(letra)
        tam = len(lista)
        if indice + avanze >= tam:
            return lista[(indice - tam) + avanze]
        return lista[indice + avanze]

    def Crearalfabeto(self):
        return list(map(chr, range(32, 254)))